### PR TITLE
[autoscaler] Ensure `run_init` happens in all cases

### PR DESCRIPTION
**Pull request by ijrsvt**
## Why are these changes needed?
There was a case where `run_init` was not executed when it should have been:
* Cluster Config hash had **not changed** 
* File Mounts has had **changed**

This isn't a full solution to #6128

## Related issue number

- Resolves more of #6128

## Checks

- [ ] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://docs.ray.io/en/latest/.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [ ] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

**Comment by ericl** [comment]
Unit test?

File: python/ray/autoscaler/updater.py [hunk 1]
```diff
@@ -294,10 +294,16 @@ def do_update(self):
         node_tags = self.provider.node_tags(self.node_id)
         logger.debug("Node tags: {}".format(str(node_tags)))
 
+        if node_tags.get(TAG_RAY_RUNTIME_CONFIG) == self.runtime_hash:
```
Comment (wuisawesome) [inline python/ray/autoscaler/updater.py:297]: Won't this always run `run_init` even when it has already been run?

Comment (ijrsvt) [reply python/ray/autoscaler/updater.py:297]: Yeah, in the case where we stop (not **terminate**) and then restart a node. This doesn't solve the problem of rerunning `setup_commands` (but those _should_ be minimal when Docker is in use).

**Review by wuisawesome — APPROVED** [review]

**Comment by ijrsvt** [comment]
@ericl I am not sure of a good way to mock out stuff at the NodeUpdater level, I think we need to add a testing suite to do so.

**Comment by ijrsvt** [comment]
@ericl Welp, added it to the regular autoscaler tests :)

**Review by ericl — APPROVED** [review]

**Comment by ericl** [comment]
Thanks for the tests

**Comment by ijrsvt** [comment]
@ericl The only failing test `test_failures.py:: est_warning_for_too_many_nested_tasks` for MacOS  is **working locally on my mac**.

**Pull request merged**